mmcv/runner/base_runner.py

```diff
@@ -523,6 +523,19 @@ def register_profiler_hook(
             hook = profiler_config
         self.register_hook(hook)
 
+    def register_npu_profiler_hook(
+        self,
+        profiler_config: Union[Dict, Hook, None],
+    ) -> None:
+        if profiler_config is None:
+            return
+        if isinstance(profiler_config, dict):
+            profiler_config.setdefault('type', 'NPUProfilerHook')
+            hook = mmcv.build_from_cfg(profiler_config, HOOKS)
+        else:
+            hook = profiler_config
+        self.register_hook(hook)
+
```

Comment:
We can use `register_custom_hooks` to register it so [this custom method](register_npu_profiler_hook) can be moved.